### PR TITLE
Add vsts ci

### PR DESCRIPTION
**Pull request by TravisEz13**
Add VSTS CI YML with tests
Verified here:  https://travisez13.visualstudio.com/powershell/powershell%20Team/_build/results?buildId=83

### CODE REVIEW

**Comment by TravisEz13** [comment]
The VSTS CI will not pass until there is a YML in master.

File: .vsts-ci/ci.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+- phase: Set_Build_Name
+#  displayName: Set Build Name
+  condition: ne(variables['Build.Reason'], 'PullRequest')
+  queue:
+    name: Hosted Ubuntu 1604
+  steps:
+  - powershell:  Write-Host "##vso[build.updatebuildnumber]$env:BUILD_SOURCEBRANCHNAME-$env:BUILD_SOURCEVERSION-$((get-date).ToString("yyyyMMddhhss"))"
+    displayName: Set Build Name for Non-PR
+    condition: ne(variables['Build.Reason'], 'PullRequest')
```
Comment (TylerLeonhardt) [inline .vsts-ci/ci.yml]: is the condition needed on L16 and L10?

File: .vsts-ci/ci.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+- template: ./phase.yml
+  parameters:
+    SnapChannel: stable
+    SnapName: powershell
+
+- template: ./phase.yml
+  parameters:
+    SnapChannel: preview
```
Comment (TylerLeonhardt) [inline .vsts-ci/ci.yml]: I thought we were publishing the `powershell-preview` package to its `stable` channel?

Comment (TylerLeonhardt) [reply .vsts-ci/ci.yml]: It seems like this is used for something else... just a switch statement. Is it needed? Couldn't we just use the SnapName?

**Review by TylerLeonhardt — APPROVED** [review]
LGTM just some questions

File: .vsts-ci/ci.yml (hunk 1 above)
Comment (TravisEz13) [reply .vsts-ci/ci.yml]: removed on L16

File: .vsts-ci/ci.yml (hunk 2 above)
Comment (TravisEz13) [reply .vsts-ci/ci.yml]: very few characters are allowed in the phase name, but it also must be unique.

Comment (TravisEz13) [reply .vsts-ci/ci.yml]: also, there is no switch statement in the language.  it would end up being a bunch of if statement.

Comment (TravisEz13) [reply .vsts-ci/ci.yml]: But we can use just the Channel... I updated the code

**Review by TylerLeonhardt — APPROVED** [review]
LGTM

**Pull request merged**